### DISCUSSION TITLE
As a brand new user coming from python I want to type "return" in my function

### DISCUSSION THREAD

**Discussion by jtwaleson** [darklang/dark#4552] [Feedback]
Trying out the platform and I instinctively type `return`. Darklang is different, but it would be amazing if the platform tells me how it works instead with a hint while typing. Or just have `return` in the language, I really like explicit return statements ;)

**Comment by pbiggar** [comment]
Hmmm, perhaps trying to type "return" should offer a tutorial or doc about how to use an expression-based language.

**Reply by jtwaleson** [reply]
Forgive my ignorance about expression based languages, but if I want to return something in the middle of a block of code, how do I do that?

**Reply by StachuDotNet** [reply]
@jtwaleson In an expression-based language, you typically _cannot_ return early (in the middle of a block of code"). Rather, you refactor your code to not require "early returns." This can feel painful at first, but generally results in much more obvious+readable+stable code. It's a bit much to fully explain within this thread but "refactoring from imperative to functional" feels like an appropriate search query to me, and yields some decent-looking results for me in YouTube. I'd be happy to externally (Discord?) work through some specific refactors, though.

**Reply by pbiggar** [reply]
I hadn't realized the jump was so big between the two. It does explain why some people have no problem at all and others are lost. Any thoughts on how/where to make this simpler (without changing the dark language, that is)?
